### ISSUE TITLE
How to use with nginx?

### ISSUE THREAD

**Issue by ghost** [kristapsdz/letskencrypt#32]
The man page goes into some detail about configuring nginx but it leaves out how to setup the ~/.well-known/ stuff for registration to work. I have tired to get this working myself without much luck.

I have the following in my nginx.conf

```
server {
  listen 80;
  server_name mydomain.com;
  include /usr/local/etc/nginx/conf/acme.conf;
}
```

Contents of acme.conf

```
location ^~ /.well-known/acme-challenge/ {
    default_type "text/plain";
    root /usr/local/www/acme;
}

location = /.well-known/acme-challenge/ {
    return 404;
}
```

Registration with acme-client fails every time.

**Comment by tokred** [comment]
Your nginx configuration is correct, at least I use the same.

How do you invoke acme-client and do you provide the correct path? Try e.g. this:
```
acme-client -mvnNC /usr/local/www/acme/.well-known/acme-challenge mydomain.com
```

Best regards

**Comment by lenlynch** [comment]
I recently set this up on OpenBSD 6.1 stable.
Verify that nginx is logging failures (in the server logs), and see if the failures provide a clue to the root cause.  My nginx wasn't configured as well as your's is, but I got it working eventually.  My working config looks very close to yours (using mostly the system defaults).

```
        location ^~ /.well-known/acme-challenge/ {
                alias /var/www/acme/;
        }
```
If you aren't getting good logs, then there are issues at the network layer, most likely.
